armi/reactor/reactors.py

```diff
@@ -111,6 +111,40 @@ def add(self, container):
                 )
             self.core = cores[0]
 
+    def incrementAssemNum(self):
+        """
+        Increase the max assembly number by one and return the current value.
+
+        Returns
+        -------
+        int
+            The new max Assembly number.
+        """
+        val = int(self.p.maxAssemNum)
+        self.p.maxAssemNum += 1
+        return val
+
+    def normalizeNames(self):
+        """
+        Renumber and rename all the Assemblies and Blocks.
+
+        This method normalizes the names in the Core then the SFP.
+
+        Returns
+        -------
+        int
+            The new max Assembly number.
+        """
+        self.p.maxAssemNum = 0
+
+        ind = self.core.normalizeNames(self.p.maxAssemNum)
+        self.p.maxAssemNum = ind
+
+        ind = self.sfp.normalizeNames(self.p.maxAssemNum)
+        self.p.maxAssemNum = ind
+
+        return ind
```

Comment:
```suggestion
        self.p.maxAssemNum = self.sfp.normalizeNames(self.core.normalizeNames(0))
        
        return self.p.maxAssemNum
```

I _think_ that does the same thing that your code currently does. It might be more confusing though lol

At the very least, I think you can avoid the initial `self.p.maxAssemNum = 0` by just doing `ind = self.core.normalizeNames(0)` instead.

Reply:
This is definitely shorter. Is it more clear or easier to read?

I don't know.

I was actually thinking of going the opposite direction, and putting a comment on line 139 and 142 to describe the system more carefully.

> OH, also, no. You _need_ to reset `self.p.maxAssemNum = 0` before you do any of this logic. Or you will get edge cases where you start numbering at a non-zero value.